src/ucp/core/ucp_ep.h

```diff
@@ -194,14 +194,18 @@ typedef struct ucp_ep_config {
  * UCP_FEATURE_STREAM specific extention of the remote protocol layer endpoint
  */
 typedef struct ucp_ep_ext_stream {
-    /* ep which owns the extension */
-    ucp_ep_h                      ucp_ep;
     /* List entry in worker's EP list */
     ucs_list_link_t               list;
-    /* Queue of receive requests posted on the EP */
-    ucs_queue_head_t              reqs;
-    /* Queue of receive descriptors with data */
-    ucs_queue_head_t              data;
+    /* ep which owns the extension */
+    ucp_ep_h                      ucp_ep;
+    /* Total buffers length queued in @ref reqs, if > 0 @ref data is not valid */
+    size_t                        reqs_buf_len;
```

Comment:
why do you need the total length? Using request counter seems to be simpler

Reply:
Right, currently this is the same because request can be completed with any amount of data <= its buffer length, but it will be useful during implementation `recv_nb` with flag `WAIT_ALL`. In this case, we can predict if we can complete such request immediately.

Reply:
I seem to miss something, but in this case you just do not need to complete request until all data is in place. Then you will decrement the request counter upon its complition